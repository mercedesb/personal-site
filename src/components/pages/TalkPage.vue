<template>
   <main class='TalkPage'>
    <PageHeader
      :color="page.color"
      :icon="page.iconSvg"
      :short="true"
    >
      <template slot='titleHeader'>
        <h1>{{page.title}}</h1>
      </template>
    </PageHeader>
    <div v-if="page.mainContent">
      <div class="PageContent">
        <ParseMarkdown class="PageContent-mainContent" :source="page.mainContent" />
        <div v-for="event in page.givenAt" :key="event.id" class="EventDetails">
          <h2 class="h4">{{event.title}}</h2>
          <div class="EventDetails-links">
            <smart-link class="EventDetails-linkItem" :to="event.slidesLink" :isExternal="true" v-if="event.slidesLink">Slides</smart-link>
            <smart-link class="EventDetails-linkItem" :to="event.videoLink" :isExternal="true" v-if="event.videoLink">Video</smart-link>
            <smart-link class="EventDetails-linkItem" :to="event.audioLink" :isExternal="true" v-if="event.audioLink">Audio</smart-link>
            <smart-link class="EventDetails-linkItem" :to="'/' + event.blogLink" v-if="event.blogLink">Blog</smart-link>
            <em>{{event.date}}</em>
          </div>
        </div>
        <br/>
      </div>
    </div>
  </main>
</template>

<script>
import PageHeader from '../PageHeader.vue'
import ParseMarkdown from '../ParseMarkdown.vue'

export default {
  components: {
    PageHeader, ParseMarkdown
  },
  props: {
    page: Object
  }
}
</script>

<style lang="scss">
  @import '../../assets/styles/variables.scss';

.TalkPage {
  @include inner-page-content;

  @include media($max-tablet) {
    display: block;
  }

  .EventDetails-links {
    .EventDetails-linkItem {
      padding: 0 $small-spacing;
      border-right: 2px solid $black;
      display: inline-block;
    }

    .EventDetails-linkItem:first-of-type {
      padding-left: 0;
    }

    .EventDetails-linkItem:last-of-type {
      border-right: none;
    }
  }
}

</style>
