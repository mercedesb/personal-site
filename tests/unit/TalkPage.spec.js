import TalkPage from "@/components/pages/TalkPage.vue";
import { TestUtility } from "./TestUtility";

describe("TalkPage", () => {
  let component;

  const initialProps = {
    page: {
      ...TestUtility.talks[0],
      iconSvg: "<svg></svg>",
      givenAt: "Apr 18 2018"
    },
    ctaLinks: []
  };

  const shallow = propsData =>
    TestUtility.shallow(TalkPage, {
      propsData: {
        ...initialProps,
        ...propsData
      }
    });

  describe("Snapshots", () => {
    it("matches snapshot", () => {
      component = shallow();
      expect(component.element).toMatchSnapshot();
    });
  });
});
