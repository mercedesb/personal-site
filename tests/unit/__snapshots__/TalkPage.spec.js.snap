// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TalkPage Snapshots matches snapshot 1`] = `
<main
  class="TalkPage"
  ctalinks=""
>
  <pageheader-stub
    color="yellow"
    icon=<svg></svg>
  >
    <template>
      <h1>
        Fun, Friendly Computer Science
      </h1>
    </template>
  </pageheader-stub>
   
  <div>
    <div
      class="PageContent"
    >
      <parsemarkdown-stub
        class="PageContent-mainContent"
        source="talk main content"
      />
       
      <div
        class="EventDetails"
      >
        <h2
          class="h4"
        >
          
        </h2>
         
        <div
          class="EventDetails-links"
        >
          <!---->
           
          <!---->
           
          <!---->
           
          <!---->
           
          <em>
            
          </em>
        </div>
      </div>
      <div
        class="EventDetails"
      >
        <h2
          class="h4"
        >
          
        </h2>
         
        <div
          class="EventDetails-links"
        >
          <!---->
           
          <!---->
           
          <!---->
           
          <!---->
           
          <em>
            
          </em>
        </div>
      </div>
      <div
        class="EventDetails"
      >
        <h2
          class="h4"
        >
          
        </h2>
         
        <div
          class="EventDetails-links"
        >
          <!---->
           
          <!---->
           
          <!---->
           
          <!---->
           
          <em>
            
          </em>
        </div>
      </div>
      <div
        class="EventDetails"
      >
        <h2
          class="h4"
        >
          
        </h2>
         
        <div
          class="EventDetails-links"
        >
          <!---->
           
          <!---->
           
          <!---->
           
          <!---->
           
          <em>
            
          </em>
        </div>
      </div>
      <div
        class="EventDetails"
      >
        <h2
          class="h4"
        >
          
        </h2>
         
        <div
          class="EventDetails-links"
        >
          <!---->
           
          <!---->
           
          <!---->
           
          <!---->
           
          <em>
            
          </em>
        </div>
      </div>
      <div
        class="EventDetails"
      >
        <h2
          class="h4"
        >
          
        </h2>
         
        <div
          class="EventDetails-links"
        >
          <!---->
           
          <!---->
           
          <!---->
           
          <!---->
           
          <em>
            
          </em>
        </div>
      </div>
      <div
        class="EventDetails"
      >
        <h2
          class="h4"
        >
          
        </h2>
         
        <div
          class="EventDetails-links"
        >
          <!---->
           
          <!---->
           
          <!---->
           
          <!---->
           
          <em>
            
          </em>
        </div>
      </div>
      <div
        class="EventDetails"
      >
        <h2
          class="h4"
        >
          
        </h2>
         
        <div
          class="EventDetails-links"
        >
          <!---->
           
          <!---->
           
          <!---->
           
          <!---->
           
          <em>
            
          </em>
        </div>
      </div>
      <div
        class="EventDetails"
      >
        <h2
          class="h4"
        >
          
        </h2>
         
        <div
          class="EventDetails-links"
        >
          <!---->
           
          <!---->
           
          <!---->
           
          <!---->
           
          <em>
            
          </em>
        </div>
      </div>
      <div
        class="EventDetails"
      >
        <h2
          class="h4"
        >
          
        </h2>
         
        <div
          class="EventDetails-links"
        >
          <!---->
           
          <!---->
           
          <!---->
           
          <!---->
           
          <em>
            
          </em>
        </div>
      </div>
      <div
        class="EventDetails"
      >
        <h2
          class="h4"
        >
          
        </h2>
         
        <div
          class="EventDetails-links"
        >
          <!---->
           
          <!---->
           
          <!---->
           
          <!---->
           
          <em>
            
          </em>
        </div>
      </div>
       
      <br />
    </div>
  </div>
</main>
`;
